go/interpreter/lfvm/stack.go

```diff
@@ -98,26 +126,40 @@ func (s *Stack) String() string {
 
 var stackPool = sync.Pool{
 	New: func() interface{} {
-		return &Stack{}
+		return &stack{}
 	},
 }
 
-func NewStack() *Stack {
-	return stackPool.Get().(*Stack)
+// NewStack returns a new stack instance from the a reuse pool. Heavy stack
+// users should use this function to prevent memory reallocation overhead.
+// This function is thread-safe.
+func NewStack() *stack {
+	return stackPool.Get().(*stack)
```

Comment:
is the number stacks in the pool a configurable value ? if not, should it be ? This is probably for a follow-up PR

Reply:
No, the way the [Pool](https://pkg.go.dev/sync#Pool) works is that every time a request is received, it checks whether there is an element. If there is none, a new one is created and handed out. If one is returned, it is added to the pool. There is no way to set a limit on the pool size. However, the implementation can automatically discard elements if there are too many.